### PR TITLE
RunInfo for WarpX

### PR DESCRIPTION
**Pull request by nruof**
This must be merged after #47 

circle ci fails in this branch, this will be fixed #47

### CODE REVIEW

File: mewarpx/mewarpx/runinfo.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+        SimInfo.nxyz (int nx, int ny, int nz)
+        SimInfo.pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax)
+        SimInfo.geom (str 'XZ', 'RZ' or 'XYZ')
+        SimInfo.dt (float)
+        SimInfo.periodic (bool)
+
+    Arguments:
+        nxyz (int, int, int): tuple of the grid dimensions
+        pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax): tuple of the
+            position limits along each dimension
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/runinfo.py]: I think for sphinx the types should be tuple, and then you could specify the types inside the tuple in the description of the parameter. ie.
```suggestion
        nxyz (tuple): tuple of ints, holding the grid dimensions
        pos_lims (tuple) : tuple of floats, holding the position limits along each 
            dimension (xmax, ymin, ymax, zmin, zmax)
```

File: mewarpx/mewarpx/runinfo.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+
+class WarpXSimInfo(SimInfo):
+
+    """Create a SimInfo object when running a warpx simulation."""
+
+    def __init__(self):
+        self.nxyz = [mwxrun.nx, mwxrun.ny, mwxrun.nz]
+        self.pos_lims = [mwxrun.xmin, mwxrun.xmax,
+                         mwxrun.ymin, mwxrun.ymax,
+                         mwxrun.zmin, mwxrun.zmax]
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/runinfo.py]: The docstring for `SimInfo` says that nxyz and pos_lims should be tuples. The definition here should be consistent with the documentation of `SimInfo`, so they should both be tuples, or both be lists.

File: mewarpx/mewarpx/runinfo.py [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+        # )
+
+
+class RunInfo(SimInfo):
+
+    """Store parameters for a WARP run.
+
+    Attributes:
+        diagnames_dict (dict): Dictionary of paths to diagnostic outputs.
+        component_labels (dict): Ordered dictionary of (name, pretty_label) for
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/runinfo.py]: I think the type for `component_labels` should be `collections.OrderedDict`.

**Review by KZhu-ME — CHANGES_REQUESTED** [review]
Just a few minor changes, but overall looks good!

**Review by roelof-groenewald — APPROVED** [review]
Looks good!

File: mewarpx/mewarpx/runinfo.py [hunk 4]
```diff
@@ -98,6 +33,15 @@ class SimInfo(object):
         SimInfo.dt (float)
         SimInfo.periodic (bool)
 
+    Arguments:
+        nxyz (list int): tuple of ints, holding the grid dimensions
+        pos_lims (list int) : tuple of floats, holding the position limits along each
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/runinfo.py]:
```suggestion
        nxyz (list): tuple of ints, holding the grid dimensions
        pos_lims (list) : tuple of floats, holding the position limits along each
```

**Review by KZhu-ME — APPROVED** [review]
LGTM

**Pull request merged**